platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/events/BaseEventHashedData.java

```diff
@@ -235,8 +230,25 @@ public void deserialize(
         otherParentHash = in.readSerializable(false, Hash::new);
         timeCreated = in.readInstant();
         in.readInt(); // read serialized length
-        transactions = in.readSerializableArray(ConsensusTransactionImpl[]::new, maxTransactionCount, true);
-        checkUserTransactions();
+
+        final TransactionConfig transactionConfig = ConfigurationHolder.getConfigData(TransactionConfig.class);
```

Comment:
ConfigurationHolder is deprecated so we want to avoid using it. Events should at some point move to protobuf, which I think will not have the option of validating them while being deserialized. So maybe this change does not make sense unfortunately 😞

Reply:
The current validation logic will need to be preserved, but if we move it to a validator that performs it on a DTO, this will be compatible with protobuf. But if its part of the serialization, we have to refactor it again afterwards